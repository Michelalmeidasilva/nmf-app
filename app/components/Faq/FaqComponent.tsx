import React, { useState } from "react";
import { TouchableOpacity, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import styles from "./FaqComponent.styles";
import Text from "../Text";

interface Props {
  title: string;
  showTopLine?: boolean;
  showBottomLine?: boolean;
  description?: string;
}

const FaqComponent: React.FC<Props> = ({ title, description }) => {
  const [isClicked, setIsClicked] = useState(false);

  return (
    <View style={styles.mainContainer}>
      <TouchableOpacity onPress={() => setIsClicked(!isClicked)} style={styles.toppomContainer}>
        {isClicked ? (
          <Ionicons
            color="#3957D0"
            style={styles.mainIcon}
            name="ios-chevron-down-outline"
            size={22}
          />
        ) : (
          <Ionicons
            color="#3957D0"
            style={styles.mainIcon}
            name="ios-chevron-up-outline"
            size={22}
          />
        )}
        <Text.Primary bold>{title}</Text.Primary>
      </TouchableOpacity>
      {isClicked && description && (
        <View style={styles.bottomContainer}>
          <Text.Tertiary>{description}</Text.Tertiary>
        </View>
      )}
    </View>
  );
};

export default FaqComponent;
