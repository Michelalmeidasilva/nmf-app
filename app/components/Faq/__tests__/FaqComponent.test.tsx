import "react-native";
import React from "react";
import { create } from "react-test-renderer";

import FaqComponent from "../FaqComponent";

describe("<FaqComponent />", () => {
  test("FaqComponent renders correctly without description", () => {
    const tree = create(<FaqComponent title={"text"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FaqComponent renders correctly with title and description", () => {
    const tree = create(<FaqComponent title={"text"} description={"LOREM IPSUM"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
