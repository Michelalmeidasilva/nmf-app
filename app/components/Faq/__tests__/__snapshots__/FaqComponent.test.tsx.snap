// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FaqComponent /> FaqComponent renders correctly with title and description 1`] = `
<View
  style={
    Object {
      "marginTop": 18,
    }
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignContent": "center",
        "borderColor": "#E7F3EB",
        "borderRadius": 10,
        "borderWidth": 1.5,
        "flexDirection": "row",
        "padding": 16,
      }
    }
  >
    <Ionicons
      color="#3957D0"
      name="ios-chevron-up-outline"
      size={22}
      style={
        Object {
          "paddingRight": 13,
          "paddingTop": 10,
        }
      }
    />
    <Text.Primary
      bold={true}
    >
      text
    </Text.Primary>
  </TouchableOpacity>
</View>
`;

exports[`<FaqComponent /> FaqComponent renders correctly without description 1`] = `
<View
  style={
    Object {
      "marginTop": 18,
    }
  }
>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignContent": "center",
        "borderColor": "#E7F3EB",
        "borderRadius": 10,
        "borderWidth": 1.5,
        "flexDirection": "row",
        "padding": 16,
      }
    }
  >
    <Ionicons
      color="#3957D0"
      name="ios-chevron-up-outline"
      size={22}
      style={
        Object {
          "paddingRight": 13,
          "paddingTop": 10,
        }
      }
    />
    <Text.Primary
      bold={true}
    >
      text
    </Text.Primary>
  </TouchableOpacity>
</View>
`;
